sw/device/lib/base/mmio.h

```diff
@@ -342,6 +342,33 @@ inline void mmio_region_write_only_set_bit32(mmio_region_t base,
   mmio_region_write_only_set_mask32(base, offset, 0x1, bit_index);
 }
 
+/**
+ * Copies a block of memory from MMIO to main memory while ensuring that MMIO
+ * accesses are word-aligned.
```

Comment:
Done.

I think `ceil(len/4) + 1` would be more accurate because there is at most 1 extra read but I used `len/4 + 2` as you suggested because it is simpler and gives the same upper bound.